Add Wegue Workshop to docs (#273)
meggsimum/wegue: docs/workshop.md

# Workshop

> This workshop uses Wegue version [`v1.0.0`](https://github.com/meggsimum/wegue/releases/tag/v0.25.0) but works for higher versions with no or little adaptation.

Reviewer: ```suggestion
This workshop uses Wegue version [`v1.0.0`](https://github.com/meggsimum/wegue/releases/tag/v1.0.0) but works for higher versions with no or little adaptation.
```

## Prerequisites

> You should have basic knowledge of using a commandline and a basic understanding of HTML, CSS and JavaScript

Reviewer: Maybe adding `Basic knowlegde of Vue.js helps but is not mandatory` would be good.

Author: done

├── WguAppTemplate.vue
└── WguApp.vue
```

For the moment we will look at the `static` directory. As the name implies, it contains static content like images, CSS files, (geo)data or configuration files. The configuration files are the central point to change the behavior of Wegue.

The file `app-conf.json` controls the Wegue instance you can see in your browser. It defines the map layers that shall be seen and also defines which features of Wegue shall be used. The default config file looks overwhelming on first sight. That's why, we replace it with a minimal configuration file,

Replace the content of `app-conf.json` with this:

```json
{

  "colorTheme": {
    "themes": {
      "light": {
        "primary": "#2E7D32"
      }
    }
  },

  "mapZoom": 2,
  "mapCenter": [0, 0],

  "mapLayers": [
    {
      "type": "XYZ",
      "name": "Carto Positron",
      "url": "https://basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
      "lid": "positron",
      "attributions": "Maps: © <a href='https://carto.com/attribution/' target='_blank'>CARTO</a> | <a href='https://openstreetmap.org/copyright' target='_blank'>© OpenStreetMap</a>-contributors | "
    }
  ]
}
```

> Open your browser that runs Wegue and refresh the page (typically clicking F5 on your keyboard). Now you should see this web map:

Reviewer: ```suggestion
Open your browser that runs Wegue and refresh the page (typically pressing F5 on your keyboard). Now you should see this web map:
```

![minimal webmap](_media/workshop/minimal-map.png)

The current color is green and defined via a RGB-code. Change it to `#E0BBE4` (a kind of purple) or any other color you like. You can use a [color picker](https://www.w3schools.com/colors/colors_picker.asp) for finding your desired RGB-code. After a refresh of your browser tab, you should see the new color.

![minimal webmap with purple color](_media/workshop/minimal-map-purple.png)

When a user opens the map it will initially show the whole world. Let's say we would like to focus our map on Frankfurt in Germany. We would need to change the two properties `mapCenter` and `mapZoom`.

> The `mapCenter` is a coordinate in used map projection. In our case the default projection "web mercator" with the code `EPSG:3857` is used. For finding the coordinate you can use [this website](https://epsg.io/map#srs=3857&x=965552.541298&y=6466228.626299&z=11&layer=streets). The result might be `965552.541298 6466228.626299`. We must convert them to a JSON Array by surrounding them with bracket `[]` and separating them by a comma `,`. Additionally, We can safely remove the numbers behind the digit, because are unnecessarily precise.

Reviewer: "The `mapCenter` is a coordinate in used map projection." => "The `mapCenter` is a coordinate in the projection of the map."

Author: fixed

The `zoomLevel` is a whole number typically between `0`(whole world) and `18` (building level). For displaying a city `12` might be a good value.

Let's apply the values to our config file. Change `mapZoom` and `mapCenter` to:

```

>   "mapZoom": 11,

Reviewer: In the text above `12` is mentioned, but here `11` is used.

Author: fixed

        "attributions": "© Bundesamt für Kartographie und Geodäsie (2022)"
      },
      {
        "type": "XYZ",
        "name": "Carto Positron",
        "url": "https://basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
        "lid": "positron",
        "isBaseLayer": true,
        "attributions": "Maps: © <a href='https://carto.com/attribution/' target='_blank'>CARTO</a> | <a href='https://openstreetmap.org/copyright' target='_blank'>© OpenStreetMap</a>-contributors | "
      }
    ]
```

## Use a module

Wegue contains a lot of different modules that can be used. A frequently needed one is a layer switcher. This enables the user to switch layers on and off. To activate it, add this snippet below to your config file. Make sure it is located on the top level and also separated by an comma from other properties:

```json
  "modules": {
      "wgu-layerlist": {
        "target": "toolbar",
        "win": "floating",
        "icon": "layers"
      }
    }
```
Let's inspect what is happening here. The property `modules` is a JSON object with currently one object. However, there can be many more. Our module is called `wgu-layerlist` and has some properties by itself as well:

> - `"target": "toolbar"` adds the button for the layertree to the toolbar, alternatively it can be added to a menu using `"target": "menu"`

Reviewer: ```suggestion
- `"target": "toolbar"` adds the button for the layer list to the toolbar, alternatively it can be added to a menu using `"target": "menu"`
```

- `"win": "floating"` tells Wegue to place the module floating over the map, alternatively `"win": "sidebar",` would place it in a sidebar
- `"icon": "layers"` defines the icon image that is displayed, it can be chosen from the same [icon font](https://fonts.google.com/icons?selected=Material+Icons&icon.query=airport) that we used earlier for the layer style

> The layer tree on the top right enables us to switch the airport layer on and off.

Reviewer: ```suggestion
The layer list on the top right enables us to switch the airport layer on and off.
```

    },
    props: {
      icon: { type: String, required: false, default: 'star' }
    }
  }
</script>

<!-- Here we do the styling of our module -->
<style scoped>
  /* our module has the class '.my-tool' and we reference it here */
  .my-tool {
    left: auto !important;
    top:300px !important;
    right: 10px;
  }
</style>
```

We need to register our modules in `app/WguAppTemplate.vue`. Add this line to the imports:

```js
import MyToolWin from './components/MyTool.vue'
```

In the same file, add the entry `'my-tool-win': MyToolWin` to the `components` property. It should look like this afterwards:

```js
  components: {
      'wgu-map': OlMap,
      'wgu-app-header': AppHeader,
      'wgu-app-footer': AppFooter,
      'wgu-app-sidebar': AppSidebar,
      'wgu-app-logo': AppLogo,
      'wgu-bglayerswitcher': BgLayerSwitcher,
      'wgu-measuretool-win': MeasureWin,
      'wgu-layerlist-win': LayerListWin,
      'wgu-helpwin-win': HelpWin,
      'wgu-infoclick-win': InfoClickWin,
      'wgu-maploading-status': MapLoadingStatus,
      'wgu-attributetable-win': AttributeTableWin,
      'wgu-maprecorder-win': MapRecorderWin,
      'my-tool-win': MyToolWin
    },
```

> The changes so far properly registered our module in Wegue. To actually activate it in our application we have to adjust our config file. Add this module to the `modules` property of your config file `app/static/app-conf.json`:

Reviewer: ```suggestion
The changes so far properly registered our module in the Wegue application. To actually activate it in our app we have to adjust our config file. Add this module to the `modules` property of your config file `app/static/app-conf.json`:
```
